internal/provisioner/provisioner_utils.go

```diff
@@ -174,3 +175,73 @@ func waitForNamespacesDeleted(ctx context.Context, namespaces []string, k8sClien
 		}
 	}
 }
+
+// createCustomErrorPagesConfigMap creates a ConfigMap with custom error pages.
+// The URL in the 403 error page is taken from the PORTAL_URL environment variable.
+func createCustomErrorPagesConfigMap(k8sClient *k8s.KubeClient, namespace string) error {
+	portalURL := os.Getenv("PORTAL_URL")
+	if portalURL == "" {
+		return errors.New("PORTAL_URL environment variable is not set")
+	}
+
+	// ConfigMap data
+	configMapData := map[string]string{
+		"404": `<!DOCTYPE html>
+<html>
+  <head><title>PAGE NOT FOUND</title></head>
+  <body>PAGE NOT FOUND</body>
+</html>`,
+		"503": `<!DOCTYPE html>
+<html>
+  <head><title>CUSTOM SERVICE UNAVAILABLE</title></head>
+  <body>CUSTOM SERVICE UNAVAILABLE</body>
+</html>`,
+		"403": fmt.Sprintf(`<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <meta http-equiv="refresh" content="0; url=%s/cloud/inaccessible" />
+    <title>HTML 5 Boilerplate</title>
+</head>
+<body>
+</body>
+</html>`, portalURL),
+	}
```

Comment:
Should we use template files embedded into the provisioner binary that can be loaded and populated here using Go's template library? It might be a bit cleaner than having HTML in a map with Sprintf?

Reply:
@stafot the portalURL configured by us internally, right?

Reply:
Right. But this PR is now obsolete as fortunately ingress-nginx approved my PR to support this functionality on their helm chart. So I will close this one and we will avoid this hackie solution. 
Thanks for taking the time and review though.